.claude/commands/hcp-generate.md

```diff
@@ -0,0 +1,490 @@
+- `--node-selector` (stringToString): Node selector for the Hosted Control Plane pods
+- `--pods-labels` (stringToString): Labels for the Hosted Control Plane pods
+- `--toleration` (stringArray): Tolerations for the hcp pods
+- `--pausedUntil` (string): Pause HostedCluster creation until specified date
+- `--wait` (bool): Block until the cluster is up
+- `--timeout` (duration): Optional timeout when using --wait
+- `--version-check` (bool): Checks version compatibility
+- `--render` (bool): Render output as YAML instead of applying
+- `--render-into` (string): Render output into specified file
+- `--render-sensitive` (bool): Include secrets in rendered output
+
+### Smart Detection of Optional Parameters
+**Only suggest optional parameters when detected in user's natural language:**
+
+**Detection Keywords → Suggested Flags:**
+- "fips", "compliance", "security" → `--fips`
+- "annotation", "metadata", "tag" → `--annotations`
+- "external dns", "loadbalancer domain", "service domain" → `--external-dns-domain`
+- "upgrade strategy", "inplace", "replace" → `--node-upgrade-type`
+- "auto repair", "self healing" → `--auto-repair`
+- "arm", "arm64", "aarch64", "multi-arch", "architecture" → `--arch` + use `-multi` release image
+- "amd64", "x86", "intel" → `--arch amd64` + use `-x86_64` release image
+- "storage class", "etcd storage" → `--etcd-storage-class`
+- "label", "labels" → `--labels`
+- "timeout", "drain timeout" → `--node-drain-timeout`
+- "techpreview", "devpreview", "feature set" → `--feature-set`
+
+**Example Smart Prompting:**
+```
+🔹 **FIPS Detected**: You mentioned security/compliance. Enable FIPS mode?
+   - --fips: [yes/no]
+
+🔹 **ARM Architecture Detected**: You mentioned ARM. Using multi-architecture configuration:
+   - --arch: arm64
+   - --release-image: quay.io/openshift-release-dev/ocp-release:4.18.0-multi
+
+🔹 **x86 Architecture Detected**: You mentioned x86/Intel. Using x86_64 configuration:
+   - --arch: amd64
+   - --release-image: quay.io/openshift-release-dev/ocp-release:4.18.0-x86_64
+```
+
+**Important**: Do NOT prompt for optional parameters unless specifically mentioned by user.
+
+## OpenShift Release Image Selection
+
+**CRITICAL**: Always include `--release-image` parameter in all commands.
+
+### Finding Supported Versions:
+For the most up-to-date list of supported OpenShift releases, check:
+**https://amd64.ocp.releases.ci.openshift.org/**
+
+### Minimum Supported Version: 4.14.x
+
+### Release Image Format:
+- **x86_64 architecture**: `quay.io/openshift-release-dev/ocp-release:X.Y.Z-x86_64`
+- **ARM64 or mixed architectures**: `quay.io/openshift-release-dev/ocp-release:X.Y.Z-multi`
+- **Even in disconnected environments**: Keep the original URL, HyperShift handles translation via image-content-sources
+
+### Architecture Detection:
+When user mentions "arm", "arm64", "aarch64", or "multi-arch" → use `-multi` release image instead of `-x86_64`
```

Comment:
We should use multi by default. This is what we are doing on ROSA HCP and ARO HCP.